families-api/app/main.py

```diff
@@ -0,0 +1,225 @@
+from typing import Generic, Optional, TypeVar
+
+from fastapi import APIRouter, Depends, FastAPI
+from pydantic_settings import BaseSettings
+from sqlalchemy import create_engine
+from sqlmodel import Field, Relationship, Session, SQLModel, func, select
+
+
+class FamilyCorpusLink(SQLModel, table=True):
+    __tablename__ = "family_corpus"  # type: ignore[assignment]
+    corpus_import_id: int | None = Field(
+        default=None, foreign_key="corpus.import_id", primary_key=True
+    )
+    family_import_id: int | None = Field(
+        default=None, foreign_key="family.import_id", primary_key=True
+    )
+
+
+class Corpus(SQLModel, table=True):
+    __tablename__ = "corpus"  # type: ignore[assignment]
+    import_id: str | None = Field(default=None, primary_key=True)
+    title: str
+    families: list["Family"] = Relationship(
+        back_populates="corpus", link_model=FamilyCorpusLink
+    )
+
+
+class FamilyGeographyLink(SQLModel, table=True):
+    __tablename__ = "family_geography"  # type: ignore[assignment]
+    geography_id: int | None = Field(
+        default=None, foreign_key="geography.id", primary_key=True
```

Comment:
quibble: I'm not au-fait with SQLModel, does having two primary keys make it a composite key? My personal preference is not to have composite primary keys, but to just have an auto-incrementing id field (i.e. id, geography_id, family_import_id)

Reply:
Very much doing document driven development here:
https://sqlmodel.tiangolo.com/tutorial/many-to-many/create-models-with-link/#link-table-model

Letting SQLModel do it's thing until it's thing isn't good enough. Which, I think until I get your OTel observability in, is good enough.